llvm/test/CodeGen/RISCV/unaligned-fp-load-store.ll

```diff
@@ -0,0 +1,189 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; RUN: llc -o - %s -target-abi l64pc128d -mtriple=riscv64-unknown-freebsd -mattr=+xcheri,+d,+cap-mode -verify-machineinstrs | FileCheck %s
```

Comment:
```suggestion
; RUN: %riscv32_cheri_purecap_llc -mattr=+f,+d -target-abi il32pc64d -verify-machineinstrs < %s \
; RUN:   | FileCheck --check-prefix=RV32IFDXCHERI %s
; RUN: %riscv64_cheri_purecap_llc -mattr=+f,+d -target-abi l64pc128d -verify-machineinstrs < %s \
; RUN:   | FileCheck --check-prefix=RV64IFDXCHERI %s
```

Reply:
It complains about a duplicate -target-abi flag if I do that. Should probably change that command line option to allow later overrides.

Reply:
Ok, then:
```suggestion
; RUN: llc -mtriple=riscv32 -mattr=+f,+d,+xcheri,+cap-mode -target-abi il32pc64d -verify-machineinstrs < %s \
; RUN:   | FileCheck --check-prefix=RV32IFDXCHERI %s
; RUN: llc -mtriple=riscv64 -mattr=+f,+d,+xcheri,+cap-mode -target-abi l64pc128d -verify-machineinstrs < %s \
; RUN:   | FileCheck --check-prefix=RV64IFDXCHERI %s
```
The RISC-V tests are very consistent about the order of the flags, using `< %s` rather than `-o - %s`, running tests for both RV32 and RV64 and using the plain `riscvXX` triple (which implies `-unknown-elf`) rather than an OS-specific one unless required. Also, D requires F (I'm surprised that doesn't warn... but it will enable it automatically for you).

Reply:
Yeah this is better, the reason I generally don't use `< %s` for crash testcases is because I can't paste the run line into a debugger with a single copy. I usually try to fix it before commit but forgot here.

Reply:
Testing RV32 exposed another crash 😢

Reply:
RV32 purecap is not happy at all if you enable d and call varargs functions in purecap.